corpdir-srv-sync/src/main/java/com/github/bordertech/corpdir/sync/common/ApiVersionKeyIdSynchronisation.java

```diff
@@ -0,0 +1,19 @@
+package com.github.bordertech.corpdir.sync.common;
+
+/**
+ * Synchronisation service for keyed API object that have version data.
+ * 
+ * @param <S> source service to load data from
+ * @param <D> destination service to write data to
+ * @param <A> the keyed API version object
+ * 
+ * @author aswinkandula
+ * @since 1.0.0
+ */
+public interface ApiVersionKeyIdSynchronisation<S, D, A> extends ApiKeyIdSynchronisation<S, D, A> {
```

Comment:
Can these generic types be made more specific?
For example (these are pseudo, not actual types)....
`A` extends `ApiIdObject`
`S` extends `ApiReadonly<A>`
`D` extends `ApiWrite<A>`